spec/outputs/datadog_metrics_spec.rb

```diff
@@ -1 +1,86 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/datadog_metrics"
+
+describe LogStash::Outputs::DatadogMetrics do
+  let(:http_instance) { double('net_http_instance') }
+  let(:request) { double('net_http_request') }
+  let(:response) { double('net_http_response') }
+
+  let(:json_obj) { double('json_obj') }
+  let(:logger) { double('logger') }
+  let(:ddtags) { ['foo', 'bar'] }
+  let(:event_hash) do
+    {
+      'message' => 'Hello world!',
+      'source' => 'datadog-metrics',
+      'type' => 'generator',
+      'host' => 'localhost',
+      '@timestamp' => LogStash::Timestamp.now
+    }
+  end
+  let(:event)  { LogStash::Event.new(event_hash) }
+  let(:params) do
+    {
+      'api_key' => '123456789',
+      'dd_tags' => ddtags,
+      'metric_type' => 'gauge'
+    }
+  end
+
+  before do
+    allow(Net::HTTP).to receive(:new).and_return(http_instance)
+    allow(Net::HTTP::Post).to receive(:new).and_return(request)
+  end
+
+  subject { described_class.new(params) }
+
+  context 'construction' do
+    it 'register method calls collaborator methods' do
+      expect(http_instance).to receive(:use_ssl=)
+      expect(http_instance).to receive(:verify_mode=)
+      subject.register
+      buffer = subject.instance_variable_get(:@buffer_config)
+      expect(buffer[:max_items]).to be 10
```

Comment:
as I mentioned before, ideally I would like the buffer to be an object not some functionality mixed in.  for me this is a matter of test granularity. I look at a public api method.  if it can take multiple arguments or formats of arguments that affect how the method behaves then I create a test for each permutation/format and set expectations that must be met to verify that the method handled its given args correctly.